src/core/src/runtime/ov_tensor.cpp

```diff
@@ -105,6 +118,119 @@ Shape Tensor::get_shape() const {
     OV_TENSOR_STATEMENT({ return _impl->getTensorDesc().getDims(); });
 }
 
+void Tensor::copy_to(const ov::Tensor& dst) const {
+    OPENVINO_ASSERT(dst.get_element_type() == get_element_type(),
+                    "Tensor element types are not equal. (src: ",
+                    get_element_type(),
+                    " != dst: ",
+                    dst.get_element_type(),
+                    ")");
+    OPENVINO_ASSERT(dst.get_shape() == get_shape(),
+                    "Tensor shapes are not equal. (src: ",
```

Comment:
I suppose `dst` tensor can be reallocated in shapes don't match. 
E.g. I can create empty tensor and copy to it:
```c++
ov::Tensor dst, src = get_src_tensor();
src.copy_to(dst);
```